src/cursor/cur_table.c

```diff
@@ -989,10 +989,14 @@ __wt_curtable_open(WT_SESSION_IMPL *session,
 
 	if (0) {
 err:		if (*cursorp != NULL) {
-			if (*cursorp != cursor)
+			if (*cursorp != cursor) {
+				/* The URI is owned by the table. */
+				(*cursorp)->internal_uri = NULL;
 				WT_TRET(__wt_cursor_close(*cursorp));
+			}
```

Comment:
I think this generally requires a great big comment.  I had to go back to understand how we could end up with `*cursorp != NULL` and `*cursorp != cursor` (i.e., a dump cursor wrapping the normal handle).

Would it be sufficient to call `(*cursorp)->close(*cursorp)` without clearing `internal_uri`?